lease.go

```diff
@@ -164,20 +170,29 @@ func updateWgPeers(lm *FileLeaseManager) error {
 	return nil
 }
 
-func (lm *FileLeaseManager) addNewPeer(pubKey string) (*WgRecord, error) {
+func (lm *FileLeaseManager) createOrUpdatePeer(email, pubKey string) (*WgRecord, error) {
 	ipnet, err := lm.findNextAvailableIpAddress()
 	if err != nil {
 		return &WgRecord{}, err
 	}
-	lm.wgRecords[pubKey] = &WgRecord{
+	lm.wgRecords[email] = &WgRecord{
+		PubKey:  pubKey,
 		IP:      ipnet.IP,
 		expires: time.Now().Add(lm.leaseTime),
 	}
+	return lm.wgRecords[email], nil
+}
+
+func (lm *FileLeaseManager) addNewPeer(email, pubKey string) (*WgRecord, error) {
+	record, err := lm.createOrUpdatePeer(email, pubKey)
+	if err != nil {
+		return &WgRecord{}, err
```

Comment:
Here (and below) should we just return `nil` on error? AFAIK it's the more common practice.